ext/cairo/rb_cairo_font_face.c

```diff
@@ -121,6 +121,53 @@ cr_font_face_quartz_supported_p (VALUE klass)
 #endif
 }
 
+#if CAIRO_HAS_FT_FONT
+static void
+handle_ft_error(FT_Error error)
+{
+#undef __FTERRORS_H__
+#define FT_ERRORDEF( e, v, s )  { e, s },
+#define FT_ERROR_START_LIST     {
+#define FT_ERROR_END_LIST       { 0, 0 } };
+  int i = 0;
+  static const struct {
+    int          code;
+    const char  *str;
+  } errors[] =
+#include FT_ERRORS_H
+  VALUE rb_eCairo_FreeType2Error = rb_define_class_under (rb_mCairo, "FreeType2Error", rb_eStandardError);
+  for (i = 0; ((unsigned int) i) < sizeof(errors) / sizeof(errors[0]); i++)
+    if (error == errors[i].code)
+      rb_raise(rb_eCairo_FreeType2Error, "FreeType2 Error: %s.", errors[i].str);
+
+  rb_raise(rb_eCairo_FreeType2Error, "FreeType2 Error: Unknown error %d.", error);
+}
+
+static VALUE
+cr_font_face_create_for_ft_face (VALUE self, VALUE path)
+{
+  FT_Library library;
+  FT_Face face;
+  FT_Error error;
+  cairo_font_face_t *cairo_face;
+
+  error = FT_Init_FreeType( &library );
+  if ( error )
+    handle_ft_error( error );
+
+  error = FT_New_Face( library, RSTRING_PTR(path), 0, &face );
+  if ( error != FT_Err_Ok )
+    handle_ft_error( error );
+
+  cairo_face = cairo_ft_font_face_create_for_ft_face ( face, 0 );
+
+  FT_Done_Face ( face );
```

Comment:
Is it OK to free the `face` here?
http://cairographics.org/manual/cairo-FreeType-Fonts.html#cairo-ft-font-face-create-for-ft-face says:

> This must be kept around until the face's ref_count drops to zero and it is freed.